modules/memory-management/src/main/java/org/mule/runtime/internal/memory/bytebuffer/DirectByteBufferProvider.java

```diff
@@ -0,0 +1,33 @@
+/*
+ * Copyright (c) MuleSoft, Inc.  All rights reserved.  http://www.mulesoft.com
+ * The software in this package is published under the terms of the CPAL v1.0
+ * license, a copy of which has been included with this distribution in the
+ * LICENSE.txt file.
+ */
+package org.mule.runtime.internal.memory.bytebuffer;
+
+import java.nio.ByteBuffer;
+
+/**
+ * A {@link org.mule.runtime.api.memory.provider.ByteBufferProvider} implementation that can be used to retrieve direct
+ * {@link ByteBuffer}'s.
+ *
+ * Based on Grizzly Implementation.
+ *
+ * @since 4.5.0
+ */
+public class DirectByteBufferProvider extends ThreadPoolBasedByteBufferProvider {
+
+  public DirectByteBufferProvider() {
+    super();
+  }
+
+  public DirectByteBufferProvider(int maxSize, int baseByteBufferSize, int growthFactor, int numberOfPools) {
+    super(maxSize, baseByteBufferSize, growthFactor, numberOfPools);
+  }
+
+  @Override
+  ByteBuffer doAllocate(int size) {
+    return ByteBuffer.allocateDirect(size);
```

Comment:
I'll hate myself for this review, but... static import

Reply:
You know this was the most enjoyable change request in this review.